File: Source/Client/NativeAndroid/MobileTasks.Android/app/src/main/java/com/magenic/mobiletasks/mobiletasksandroid/LoginActivity.java
package com.magenic.mobiletasks.mobiletasksandroid;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

import com.google.common.util.concurrent.FutureCallback;
import com.google.common.util.concurrent.Futures;
import com.google.common.util.concurrent.ListenableFuture;
import com.magenic.mobiletasks.mobiletasksandroid.constants.NetworkConstants;
import com.magenic.mobiletasks.mobiletasksandroid.models.Login;
import com.microsoft.windowsazure.mobileservices.MobileServiceActivityResult;
import com.microsoft.windowsazure.mobileservices.authentication.MobileServiceAuthenticationProvider;
import com.microsoft.windowsazure.mobileservices.authentication.MobileServiceUser;
import com.microsoft.windowsazure.mobileservices.MobileServiceException;

public class LoginActivity extends ActivityBase implements View.OnClickListener {

    private Login login;
    public static final int LOGIN_REQUEST_CODE = 1;
    private MobileServiceAuthenticationProvider requestedProvider;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.login);

        login = new Login(this, networkService);

        ImageButton button = (ImageButton) this.findViewById(R.id.facebook);
        button.setOnClickListener(this);
        button = (ImageButton) this.findViewById(R.id.google);
        button.setOnClickListener(this);
        button = (ImageButton) this.findViewById(R.id.microsoft);
        button.setOnClickListener(this);
        button = (ImageButton) this.findViewById(R.id.twitter);
        button.setOnClickListener(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (login.checkPreviousAuthentication()) {
            startActivity(new Intent(this, MainActivity.class));
        }
    }

    @Override
    public void onClick(View v) {
        MobileServiceAuthenticationProvider provider = MobileServiceAuthenticationProvider.MicrosoftAccount;
        switch (v.getId()) {
            case R.id.facebook:
                provider = MobileServiceAuthenticationProvider.Facebook;
                break;
            case R.id.google:
                provider = MobileServiceAuthenticationProvider.Google;
                break;
            case R.id.microsoft:
                provider = MobileServiceAuthenticationProvider.MicrosoftAccount;
                break;
            case R.id.twitter:
                provider = MobileServiceAuthenticationProvider.Twitter;
                break;
        }
        requestedProvider = provider;

        login.authenticate(provider, LOGIN_REQUEST_CODE);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        // When request completes
        if (resultCode == RESULT_OK) {
            // Check the request code matches the one we send in the login request
            if (requestCode == LOGIN_REQUEST_CODE) {
                MobileServiceActivityResult result = this.networkService.getClient().onActivityResult(data);
                if (result.isLoggedIn()) {
                    login.setToken(requestedProvider);
                    startActivity(new Intent(this, MainActivity.class));
                } else {
                    // login failed, check the error message

                }
            }
        } else {
            this.showMessage("Login Failure", "Error logging in");
        }
    }
}

File: Source/Client/Xamarin.Android/MobileTasks.Android/MobileTasks.Android/Activities/LoginActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using MobileTasks.Droid.Services;
using Microsoft.WindowsAzure.MobileServices;
using System.Threading.Tasks;

namespace MobileTasks.Droid.Activities
{
	[Activity(Label = "Login", MainLauncher = true, Icon = "@drawable/icon")]
	public class LoginActivity : Activity
	{
		protected override void OnCreate(Bundle savedInstanceState)
		{
			base.OnCreate(savedInstanceState);

			SetContentView(Resource.Layout.Login);

			var facebook = FindViewById<ImageButton>(Resource.Id.facebook);
			facebook.Click += async delegate { await AuthenticateAsync(MobileServiceAuthenticationProvider.Facebook); };

			var microsoft = FindViewById<ImageButton>(Resource.Id.microsoft);
			microsoft.Click += async delegate { await AuthenticateAsync(MobileServiceAuthenticationProvider.MicrosoftAccount); };

			var twitter = FindViewById<ImageButton>(Resource.Id.twitter);
			twitter.Click += async delegate { await AuthenticateAsync(MobileServiceAuthenticationProvider.Twitter); };

			var google = FindViewById<ImageButton>(Resource.Id.google);
			google.Click += async delegate { await AuthenticateAsync(MobileServiceAuthenticationProvider.Google); };
		}

		private async Task AuthenticateAsync(MobileServiceAuthenticationProvider provider)
		{
			try
			{
				await MobileService.Instance.LoginAsync(this, provider);

				var intent = new Intent(this, typeof(MainActivity));
				StartActivity(intent);
			}
			catch (InvalidOperationException)
			{
				// TODO: Show an error;
				var blah = 0;
			}
		}
	}
}